### PR TITLE
Adds `Read` instances for `Size` and `Seed`.

### PR DESCRIPTION
**Pull request by dalaing**
These should match the `Show` instances.

### CODE REVIEW

**Comment by charleso** [comment]
I know testing has been thin on the ground while Jacob has been smashing things out, but should we add a round-trip read/show property test for these? :)

**Comment by dalaing** [comment]
That's a very good call.

Also, while I'm at it: I'm guessing I should write the `Read` instances that correspond to the other `Show` instances in `hedgehog`, rather than just doing what I need and floating on to the next thing :)

**Comment by jystic** [comment]
> Also, while I'm at it: I'm guessing I should write the Read instances that correspond to the other Show instances in hedgehog, rather than just doing what I need and floating on to the next thing :)

Only if you actually need them, I'd be ok with only having `Read` for `Size` and `Seed`, the other stuff probably won't end up getting rendered anywhere.

File: hedgehog/test/Text.hs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  a <- forAll g
+  elem (a, "") (readsPrec p (showsPrec p a "")) === True
+
+textProperties :: [Property]
+textProperties =
+  [ showAppendProp genSize
+  , readShowProp genSize
+  , showAppendProp genSeed
+  , readShowProp genSeed
+  ]
```
Comment (jystic) [inline hedgehog/test/Text.hs]: You might as well use `$((checkAll))` here (just need to name the properties starting with `prop_`)

Comment (jystic) [inline hedgehog/test/Text.hs]: oh sorry, I see what you've done, yeah fair enough

File: hedgehog/test/Text.hs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+module Text (
+    textProperties
+  ) where
+
+import           Data.Foldable (elem)
```
Comment (jystic) [inline hedgehog/test/Text.hs]: I don't think this is required, seems to be breaking the build for 7.10:
```
test/Text.hs:5:1: Warning:
    The import of ‘Data.Foldable’ is redundant
      except perhaps to import instances from ‘Data.Foldable’
    To import instances alone, use: import Data.Foldable()
```

**Comment by jystic** [comment]
Sorry to be a pain, I haven't got a build set up to check this yet, but would you mind rebasing this on top the latest master? Github doesn't have a rebase + merge --no-ff button which is how I want this repo work.

**Comment by jystic** [comment]
Have rebased and merged

**Pull request closed without merging**